libsel4/include/sel4/bootinfo_types.h

```diff
@@ -24,11 +24,12 @@ enum {
     seL4_CapDomain              = 11, /* global domain controller cap */
     seL4_CapSMMUSIDControl      = 12,  /*global SMMU SID controller cap, null cap if not supported*/
     seL4_CapSMMUCBControl       = 13,  /*global SMMU CB controller cap, null cap if not supported*/
+    seL4_CapSMC                 = 14,  /*global SMC cap, null cap if not supported*/
```

Comment:
Is this a potentially breaking change that should be noted in the readme?

Reply:
That is a good point I hadn't considered. Adding to the end avoids breaking old layouts, but that becomes much harder to do with the sched_context cap getting configured out. Maybe positions should be constant whether their features are enabled or not? 
ex:
```
enum {
    seL4_CapNull                =  0, /* null cap */
    seL4_CapInitThreadTCB       =  1, /* initial thread's TCB cap */
    seL4_CapInitThreadCNode     =  2, /* initial thread's root CNode cap */
    seL4_CapInitThreadVSpace    =  3, /* initial thread's VSpace cap */
    seL4_CapIRQControl          =  4, /* global IRQ controller cap */
    seL4_CapASIDControl         =  5, /* global ASID controller cap */
    seL4_CapInitThreadASIDPool  =  6, /* initial thread's ASID pool cap */
    seL4_CapIOPortControl       =  7, /* global IO port control cap (null cap if not supported) */
    seL4_CapIOSpace             =  8, /* global IO space cap (null cap if no IOMMU support) */
    seL4_CapBootInfoFrame       =  9, /* bootinfo frame cap */
    seL4_CapInitThreadIPCBuffer = 10, /* initial thread's IPC buffer frame cap */
    seL4_CapDomain              = 11, /* global domain controller cap */
    seL4_CapSMMUSIDControl      = 12,  /*global SMMU SID controller cap, null cap if not supported*/
    seL4_CapSMMUCBControl       = 13,  /*global SMMU CB controller cap, null cap if not supported*/
    seL4_CapInitThreadSC        = 14, /* initial thread's scheduling context cap */
    seL4_CapSMC                    = 15,  /*global SMC cap, null cap if not supported*/
    seL4_NumInitialCaps         = 16
};
```

Then if they feature isn't enabled, it just shows up as a nullcap?

Reply:
@lsf37, @kent-mcleod; What do you think about this? 

A change here might also require updating `SIMPLE_SKIPPED_INIT_CAPS` here:
- https://github.com/seL4/seL4_libs/blob/master/libsel4simple/arch_include/arm/simple/arch/simple.h#L17-L19 (note that there is no RISC-V here, it uses the ARM files)
- https://github.com/seL4/seL4_libs/blob/master/libsel4simple/arch_include/x86/simple/arch/simple.h#L17-L23

Reply:
@lsf37, @kent-mcleod; ping.

Reply:
@lsf37, @kent-mcleod; pinging again.

Reply:
Apologies for the long silence. As commented in #1088, from the verification side that would be Ok.

Reply:
Once this is rebased on the changes from #1088, this should be resolved?